sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/client.py

```diff
@@ -105,27 +105,28 @@ class AMQPClient(object):
     :type handle_max: int
     :param on_attach: A callback function to be run on receipt of an ATTACH frame.
      The function must take 4 arguments: source, target, properties and error.
-    :type on_attach: func[~uamqp.address.Source, ~uamqp.address.Target, dict, ~uamqp.errors.AMQPConnectionError]
+    :type on_attach: func[~pyamqp.endpoint.Source, ~pyamqp.endpoint.Target, dict, ~pyamqp.error.AMQPConnectionError]
     :param send_settle_mode: The mode by which to settle message send
      operations. If set to `Unsettled`, the client will wait for a confirmation
      from the service that the message was successfully sent. If set to 'Settled',
      the client will not wait for confirmation and assume success.
-    :type send_settle_mode: ~uamqp.constants.SenderSettleMode
+    :type send_settle_mode: ~pyamqp.constants.SenderSettleMode
     :param receive_settle_mode: The mode by which to settle message receive
      operations. If set to `PeekLock`, the receiver will lock a message once received until
      the client accepts or rejects the message. If set to `ReceiveAndDelete`, the service
      will assume successful receipt of the message and clear it from the queue. The
      default is `PeekLock`.
-    :type receive_settle_mode: ~uamqp.constants.ReceiverSettleMode
+    :type receive_settle_mode: ~pyamqp.constants.ReceiverSettleMode
     :param encoding: The encoding to use for parameters supplied as strings.
      Default is 'UTF-8'
     :type encoding: str
     """
 
-    def __init__(self, hostname, auth=None, **kwargs):
-        self._hostname = hostname
+    def __init__(self, remote_address, auth=None, client_name=None, debug=False, retry_policy=None, 
+    keep_alive_interval=None, **kwargs):
+        self._remote_address = remote_address.address if (isinstance(remote_address, Source) or isinstance(remote_address, Target)) else remote_address
```

Comment:
we format hostname in the EH SDK side - will need to do something similiar in pyamqp to remote_address otherwise wont be able to connect to socket